zaza/openstack/charm_tests/keystone/tests.py

```diff
@@ -371,3 +370,79 @@ def test_security_checklist(self):
             expected_passes,
             expected_failures,
             expected_to_pass=False)
+
+
+class LdapTests(BaseKeystoneTest):
+    """Keystone ldap tests tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running Keystone ldap-tests."""
+        super(LdapTests, cls).setUpClass()
+
+    def _get_ldap_config(self):
+        """Generate ldap config for current model.
+
+        :return: tuple of whether ldap-server is running and if so, config
+            for the keystone-ldap application.
+        :rtype: Tuple[bool, Dict[str,str]]
+        """
+        ldap_ips = zaza.model.get_app_ips("ldap-server")
+        self.assertTrue(ldap_ips, "Should be at least one ldap server")
+        return {
+            'ldap-server': "ldap://{}".format(ldap_ips[0]),
+            'ldap-user': 'cn=admin,dc=test,dc=com',
+            'ldap-password': 'crapper',
+            'ldap-suffix': 'dc=test,dc=com',
+            'domain-name': 'userdomain',
+        }
+
+    def _find_keystone_v3_user(self, username, domain):
+        """Find a user within a specified keystone v3 domain.
+
+        :param str username: Username to search for in keystone
+        :param str domain: username selected from which domain
+        :return: return username if found
+        :rtype: Optional[str]
+        """
+        client = self.admin_keystone_client
+        domain_users = client.users.list(
+            domain=client.domains.find(name=domain).id
+        )
+        usernames = []
+        for user in domain_users:
+            usernames.append(user.name)
+            if username.lower() == user.name.lower():
+                return user
+        logging.debug(
+            "User {} was not in these users: {}. Returning None."
+            .format(username, usernames)
+        )
+        return None
+
+    def test_100_keystone_ldap_users(self):
+        """Validate basic functionality of keystone API with ldap."""
+        application_name = 'keystone-ldap'
+        config = self._get_ldap_config()
+
+        with self.config_change(
+                self.config_current(application_name),
+                config,
+                application_name=application_name):
+            logging.info(
+                'Waiting for users to become available in keystone...'
+            )
+            states = {
+                'keystone': {
+                    'workload-status': 'idle',
```

Comment:
There is another method for checking for idleness. workload-state should be active here. This is, however, the default, making the setting of states unecessary.
The best practice is to check the tests.yaml which allows future developers to change expectations:
        test_config = lifecycle_utils.get_charm_config(fatal=False)
        zaza.model.wait_for_application_states(
            states=test_config.get("target_deploy_status", {}))

Reply:
did this!